python/pyarrow/tests/test_dataset.py

```diff
@@ -78,9 +79,18 @@ def _table_from_pandas(df):
     return table.replace_schema_metadata()
 
 
+def _filesystem_uri(path):
+    # URIs on Windows must follow 'file:///C:...' or 'file:/C:...' patterns.
+    if isinstance(path, pathlib.WindowsPath):
```

Comment:
Rather `if os.name == 'nt'`. This function should ideally work fine on regular string.